api/app/attendance/api.py

```diff
@@ -85,8 +113,53 @@ def post(self):
                 event_name=event.name)
         )
 
-        return attendance, 201
-
+        # Other Fields
+        registration = None
+        offer = db.session.query(Offer).filter(
+            Offer.user_id == 1).filter(Offer.event_id == event_id).first()
+
+        if not offer:
+            # Check if Guest Registration
+            registration_form = db.session.query(RegistrationForm).filter(
+                RegistrationForm.event_id == event_id).first()
+            if(registration_form):
+                registration = db.session.query(GuestRegistration).filter(
+                    GuestRegistration.user_id == user_id).filter(GuestRegistration.registration_form_id == registration_form.id).first()
+
+        else:
+            # Normal registration
+            registration = db.session.query(Registration).filter(
+                Registration.offer_id == offer.id).first()
+
+        unavilable_response = None
+        if(registration is None):
+            is_guest_registration = unavilable_response
+            has_accepted_accom_award = unavilable_response
+            role = unavilable_response
+        else:
+            is_guest_registration = (not offer and registration)
+            event_role = db.session.query(EventRole).filter_by(

```

Comment:
Sorry, this was meant to be the invited guest role (which you can find on the invited_guest table) - like "Speaker" or "Tutor", "Organiser" etc - this is mostly because Tutors and Organisers receive different T-shirts

Reply:
Cool, I have changed this as well. Just confirming, are all tutors invited guests?

Reply:
Thanks, looks good! 
So, one complication is that some people have offers AND are invited guests - most of the tutors fall in this category. So we should try to read the invited guest role irrespective of whether they have an offer or not.